src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -108,72 +115,95 @@ public bool TryUnparse(StringBuilder sb)
+            /// </summary>
+            /// <param name="input">Name of the input column.</param>
+            /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced. </param>
+            /// <param name="kind">Whitening kind (PCA/ZCA).</param>
+            /// <param name="eps">Scaling regularizer.</param>
+            /// <param name="maxRows">Max number of rows.</param>
+            /// <param name="saveInverse">Whether to save inverse (recovery) matrix.</param>
+            /// <param name="pcaNum">PCA components to retain.</param>
+            public ColInfo(string input, string output = null, WhiteningKind kind = Defaults.Kind, float eps = Defaults.Eps,
+                int maxRows = Defaults.MaxRows, bool saveInverse = Defaults.SaveInverse, int pcaNum = Defaults.PcaNum)
             {
+                Input = input;
+                Contracts.CheckValue(Input, nameof(Input));
+                Output = output ?? input;
+                Contracts.CheckValue(Output, nameof(Output));
+                Kind = kind;
+                Contracts.CheckUserArg(Kind == WhiteningKind.Pca || Kind == WhiteningKind.Zca, nameof(Kind));
+                Epsilon = eps;
+                Contracts.CheckUserArg(0 <= Epsilon && Epsilon < float.PositiveInfinity, nameof(Epsilon));
+                MaxRow = maxRows;
+                Contracts.CheckUserArg(MaxRow > 0, nameof(MaxRow));
+                SaveInv = saveInverse;
+                PcaNum = pcaNum;
+                Contracts.CheckUserArg(PcaNum >= 0, nameof(PcaNum));
+            }
+
+            internal ColInfo(Column item, Arguments args)
+            {
+                Input = item.Source;
+                Contracts.CheckValue(Input, nameof(Input));
+                Output = item.Name ?? item.Source;
+                Contracts.CheckValue(Output, nameof(Output));
                 Kind = item.Kind ?? args.Kind;
                 Contracts.CheckUserArg(Kind == WhiteningKind.Pca || Kind == WhiteningKind.Zca, nameof(item.Kind));
                 Epsilon = item.Eps ?? args.Eps;
-                Contracts.CheckUserArg(0 <= Epsilon && Epsilon < Float.PositiveInfinity, nameof(item.Eps));
+                Contracts.CheckUserArg(0 <= Epsilon && Epsilon < float.PositiveInfinity, nameof(item.Eps));
                 MaxRow = item.MaxRows ?? args.MaxRows;
                 Contracts.CheckUserArg(MaxRow > 0, nameof(item.MaxRows));
                 SaveInv = item.SaveInverse ?? args.SaveInverse;
                 PcaNum = item.PcaNum ?? args.PcaNum;
                 Contracts.CheckUserArg(PcaNum >= 0, nameof(item.PcaNum));
-
-                if (Kind == WhiteningKind.Zca || PcaNum == 0)
-                    Type = info.TypeSrc.AsVector;
-                else
-                    Type = new VectorType(NumberType.Float, PcaNum); // REVIEW: make it work with pcaNum == 1.
             }
 
```

Comment:
Can those checks be removed?

Reply:
You are right the problem persists, I will keep that. (Assuming you are talking about the REVIEW comment)

---
In reply to: [229526809](https://github.com/dotnet/machinelearning/pull/1452#discussion_r229526809) [](ancestors = 229526809)